packages/acs-ui-javascript-loaders/src/callCompositeLoader.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React from 'react';
+
+const reactVersion = React.version;
+const parseReactVersion = (version: string | undefined): number[] => {
+  if (!version) {
+    return [];
+  }
+  return version.split('.').map((v) => parseInt(v));
+};
+if (parseReactVersion(reactVersion)[0] && parseReactVersion(reactVersion)[0] < 18) {
+  throw new Error(
+    'React version is less than 18. Please upgrade to React 18 or alternatively checkout how to use our composites directly here: https://azure.github.io/communication-ui-library/?path=/docs/quickstarts-composites--page'
+  );
+}
+
+import { createRoot } from 'react-dom/client';
+import { AzureCommunicationTokenCredential, CommunicationUserIdentifier } from '@azure/communication-common';
+import { fromFlatCommunicationIdentifier } from '@internal/acs-ui-common';
+import {
+  CallComposite,
+  createAzureCommunicationCallAdapter,
+  AzureCommunicationCallAdapterOptions,
+  CallAdapter,
+  CallAdapterLocator,
+  CallCompositeOptions
+} from '@internal/react-composites';
+import { initializeIcons } from '@fluentui/react';
+
+/**
+ * Props for the OutboundCallComposite that you can use in your application.
+ * @public
+ */
+export type CallCompositeLoaderProps = {
+  userId: string;
+  token: string;
+  displayName: string;
+  locator: CallAdapterLocator;
+  options?: AzureCommunicationCallAdapterOptions;
+};
+
+/**
+ * Loader function for the OutboundCallComposite that you can use in your application.
+ *
+ * @public
+ */
+export const loadCallComposite = async function (
```

Comment:
does not have to be part of this PR:

I feel there is some common logic could be share between these 2 loaders - people can just provide args and create...Adapter functions to make a new loader function (I guess there would be chat loader and more)

we might be able to have a function internally - createCompositeLoader(CompositeFunction, createAdapterFunction) => async function(args, html, props)

To make it easy for warpping new composite into wrapper

Reply:
this is true, we could couple it this way since it isn't a JS bundle. this pattern is followed in the PR from a hold over from the need for specific params needed but TS saves us here. will keep it in mind for the future work